boards/DCM/Src/App/torquevectoring/App_TorqueVectoring.c

```diff
@@ -0,0 +1,97 @@
+#include "torquevectoring/App_TorqueVectoring.h"
+#include "torquevectoring/App_TorqueVectoringConstants.h"
+#include "torquevectoring/App_PowerLimiting.h"
+#include "torquevectoring/App_ActiveDifferential.h"
+#include "torquevectoring/App_TractionControl.h"
+#include "App_Timer.h"
+#include "App_CanRx.h"
+#include "App_CanTx.h"
+
+TimerChannel pid_timeout;
+
+PowerLimiting_Inputs       power_limiting_inputs;
+ActiveDifferential_Inputs  active_differential_inputs;
+ActiveDifferential_Outputs active_differential_outputs;
+TractionControl_Inputs     traction_control_inputs;
+TractionControl_Outputs    traction_control_outputs;
+
+PID pid_power_correction;
+float pid_power_correction_factor = 1.0f;
+
+void App_TorqueVectoring_Setup(void) {
+    App_PID_Init(&pid_power_correction, &PID_POWER_CORRECTION_CONFIG);
+
+    PID pid_traction_control;
+    App_PID_Init(&pid_traction_control, &PID_TRACTION_CONTROL_CONFIG);
+    traction_control_inputs.pid = &pid_traction_control;
+
+    App_Timer_InitTimer(&pid_timeout, PID_TIMEOUT_ms);
+}
+
+void App_TorqueVectoring_Run(void)
+{
+    // FIXME: Replace with CAN messages
+    float accelerator_pedal_percent = 2.0;
+    bool regen_engaged = true;
+
+    if (accelerator_pedal_percent > 0.0f) {
+        // Reset control loops if timeout elapsed
+        TimerState timeout = App_Timer_UpdateAndGetState(&pid_timeout);
+        if (timeout == TIMER_STATE_EXPIRED) {
+            App_PID_RequestReset(&pid_power_correction);
+            pid_power_correction_factor = 1.0f;
+            App_PID_RequestReset(traction_control_inputs.pid);
+
+        }
+        App_Timer_Restart(&pid_timeout);
```

Comment:
Imo the code will be easier to read + maintain if you only handle the state machine logic between regen, torque vectoring, and zero torque here. And then call other functions in this file that handle what each state actually does. 
Right now it's a little tricky to read what code is for the algorithm and what code is for the state machine, which I'd imagine will only get worse as the algorithms get more complicated.